cdap-app-fabric/src/main/java/co/cask/cdap/app/store/RuntimeStore.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.app.store;
+
+import co.cask.cdap.api.workflow.Workflow;
+import co.cask.cdap.api.workflow.WorkflowToken;
+import co.cask.cdap.app.program.Program;
+import co.cask.cdap.proto.BasicThrowable;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramRunStatus;
+import co.cask.cdap.proto.WorkflowNodeStateDetail;
+import co.cask.cdap.proto.id.ProgramRunId;
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * A store for runtime information of a {@link Program}.
+ */
+public interface RuntimeStore {
+
+  /**
+   * Compare and set operation that allow to compare and set expected and update status.
+   * Implementation of this method should guarantee that the operation is atomic or in transaction.
+   *
+   * @param id              Info about program
```

Comment:
this may just be copied over, but let's fix the javadocs in this class to have a single space between parameter name and description.

Also, this is the program Id, not info about program?